doc/md/guides/mysql/cd-101.md

```diff
@@ -0,0 +1,106 @@
+---
+
+## Introduction
+
+This document describes in detail the detection of foreign keys drops for MySQL databases, 
+using Atlas's linting capabilities.
+
+A foreign key constraint can be dropped from a table using the following command:
+```sql
+ALTER TABLE pets DROP FOREIGN KEY owner_id;
+```
+
+Even though this action is not causing data loss and there might be good reasons to perform it on a production database, it can have negative and sometimes unpredictable side effects (some of them are explored below) which should be taken into consideration.
+Atlas's linter can be used to raise a flag when a foreign key drop is about to be executed. Simply run:
+```bash
+atlas migrate lint --dir file://migrations --dev-url "mysql://root:pass@localhost:3306/dev" --latest 1
+```
+The `--latest` flag tells atlas to inspect only the last migration. If the last migration file in your migrations directory contains a foreign key drop command, the last command will output something like:
+
+```bash
+20230305125128.sql: constraint deletion detected:
+
+	L1: Dropping foreign-key constraint "owner_id"
+```
+
+[Click here](/versioned/lint) for more information about how to use the Atlas CLI Linter.
+
+## Impact of dropping foreign key constraints
+
+### Data integrity
+Like mentioned above, foreign key drop is not a destructive action, meaning it's not causing data loss. But it does have an impact on the **structure** of the data. In other words, this action won't change the data in the table, but it will change the way different tables interact with one another.
+
+Suppose we have the following database:
+<p style={{textAlign: "center"}}><a href="https://gh.atlasgo.cloud/explore/e3d188d0"><img src="https://atlasgo.io/uploads/users-pets.png" alt="Pets ERD"/></a></p>
+
+ And we drop the foreign key between `pets` and `users` with the following command: 
+
+```sql
+ALTER TABLE pets DROP FOREIGN KEY owner_fk;
+```
+
+Now, inserting data into the `pets` is still possible as before. But it is also possible to insert "invalid" data - a pet with an owner_id which doesn't exist in the `users` table. If such data is inserted, queries that relies on the foreign key to validate the data can fail.
+
+
+### Irreversible changes
+Dropping a foreign key constraint is not always a reversible operation. If data which would have been blocked by the constraint was inserted to the table, then the constraint cannot be re-applied without deleting the data.
+
+In the example above, trying to re-create the constraint with:
+```sql
+ALTER TABLE pets ADD CONSTRAINT owner_id FOREIGN KEY (owner_id) REFERENCES users (id);
+```
+may result in the following error:
+```
+(1452, 'Cannot add or update a child row: a foreign key constraint fails (`pets`.`#sql-1_f83`, CONSTRAINT `owner_id` FOREIGN KEY (`owner_id`) REFERENCES `users` (`id`))')
+```
+
+The only way the constraint could be applied again is by deleting all the invalid rows from the "pets" table.
+
```

Comment:
"Show by example": delete the invalid row and then try again and see it works.

Reply:
I thought about it, but it would complicate the example. I'll have to show the invalid data beforehand so I can then show what I'm deleting. I think it'll make the example messier.